### PR TITLE
PROD-406: telegram-sign-in-2a

### PR DESCRIPTION
**Pull request by yash431garg**
- Description
1. Save user telegram username to database
2. Access PWA in mini app with answer question, reveal, burn and claim flow.
- Screen shots or recordings for UI changes
N/A

### CODE REVIEW

File: app/login/page.tsx [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+  //   searchParams,
+  // }: {
+  //   searchParams: { [key: string]: string | string[] | undefined };
+  // }
   const [dailyDeck, payload] = await Promise.all([
     getDailyDeck(),
     getJwtPayload(),
   ]);
 
+  // const telegramAuthToken = searchParams.telegramAuthToken as string;
```
Comment (marvinmarnold) [inline app/login/page.tsx:28]: Can you remove unused code?

File: app/screens/LoginScreens/LoginScreen.tsx [hunk 2]
```diff
@@ -1,66 +1,98 @@
...
+          walletConnectorName: primaryWallet?.connector?.name,
+        });
+
+      setIsLoading(false);
+    }
+
+    if (!!payload?.sub && !!authToken && awaitingSignatureState === "idle") {
+      const destination = params.get("next");
+      if (!!destination) {
+        console.log("redirect");
```
Comment (marvinmarnold) [inline app/screens/LoginScreens/LoginScreen.tsx:80]: can remove

**Review by marvinmarnold — APPROVED** [review]
This looks good. Am I understanding that you want to follow up in a different issue to?
1. Get tg mini app login working from main /login path
2. Miniapp working in TG web

**Review by neel-ds — APPROVED** [review]

**Comment by neel-ds** [comment]
LGTM!

**Pull request merged**